TTS/tts/multispeaker/adverserial_classifier.py

```diff
@@ -0,0 +1,54 @@
+import torch
+from torch.nn import functional as F
+from torch.nn import Sequential, Linear
+
+
+class GradientReversalFunction(torch.autograd.Function):
+    """Revert gradient without any further input modification."""
+
+    @staticmethod
+    def forward(ctx, x, l, c):
+        ctx.l = l
+        ctx.c = c
+        return x.view_as(x)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        grad_output = grad_output.clamp(-ctx.c, ctx.c)
+        return ctx.l * grad_output.neg(), None, None
+
+class ReversalClassifier(torch.nn.Module):
```

Comment:
You may want to move this file to: TTS/tts/layers/tacotron/adverserial_classifier.py
I believe it is more compatible with the current style :).

Reply:
Sure